chainerx_cc/CMakeLists.txt

```diff
@@ -65,13 +73,66 @@ if(NOT ((CMAKE_BUILD_TYPE STREQUAL "Release") OR (CMAKE_BUILD_TYPE STREQUAL "Deb
     message(FATAL_ERROR "Invalid CMAKE_BUILD_TYPE: " ${CMAKE_BUILD_TYPE})
 endif()
 
-# CUDA
-if(${CHAINERX_BUILD_CUDA})
+#-----------
+# C++ setup
+#-----------
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
Seems some compilation error is caused by the move of this.

Reply:
Fixed (hopefully): moved `list(APPEND CUDA_NVCC_FLAGS -std=c++14)`

Reply:
I'd like [`CUDA_STANDARD`](https://cmake.org/cmake/help/v3.8/prop_tgt/CUDA_STANDARD.html#prop_tgt:CUDA_STANDARD) more if we can use cmake with its support

Reply:
`CUDA_NVCC_FLAGS` seems optimal for now since current cmake requirement is 3.1